### PR TITLE
Make target folder before downloading the sample

### PR DESCRIPTION
**Pull request by jonasem**

### CODE REVIEW

File: Changelog.md [hunk 1]
```diff
@@ -4,6 +4,11 @@
 
 -   Option to auto detect trace database when reading trace file.
 
+### Fixed
+
+-   Programming sample on windows sometimes crashed when the target folder was
+    missing.
```
Comment (aadnekar) [inline Changelog.md:10]:
```suggestion
-   Programming sample could fail the first time programming an application.
```

File: src/features/programSample/samples.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
 import { existsSync } from 'fs';
 import { copyFile, mkdir, readFile, writeFile } from 'fs/promises';
-import { dirname, join } from 'path';
+import { join } from 'path';
```
Comment (aadnekar) [inline src/features/programSample/samples.ts:9]: Why is this added in this commit, cannot see it being used in the diff.

File: src/features/programSample/samples.ts [hunk 3]
```diff
@@ -62,8 +62,13 @@ export const downloadSampleIndex = () =>
 export const downloadModemFirmware = (modemFirmware: ModemFirmware) =>
     downloadFile(modemFirmware.file);
 
-export const downloadSample = (sample: Sample) =>
+export const downloadSample = async (sample: Sample) => {
+    if (!existsSync(DOWNLOAD_FOLDER)) {
+        await mkdir(DOWNLOAD_FOLDER);
```
Comment (aadnekar) [inline src/features/programSample/samples.ts:67]: Why not just use `mkdirSync`?

**Review by aadnekar — APPROVED** [review]
Looks good, added a couple of comments.

File: src/features/programSample/samples.ts (hunk 3 above)
Comment (jonasem) [reply src/features/programSample/samples.ts:67]: That would block the thread?

Comment (datenreisender) [reply src/features/programSample/samples.ts:67]: Isn't the same true for `existsSync` above? Either we say that these FS operations are fast enough and it doesn't matter in this place, then we can make both sync or we should make both async.

(And generally we should move all of these low level operations out of the renderer thread because they are critical there anyhow, but that is another thing.)

Comment (jonasem) [reply src/features/programSample/samples.ts:67]: Yeah, there is no `exists` in the fs/promises, but I guess the optimal way would be to use `access` or something similar to avoid the locking.
